src/main/java/io/stargate/sgv2/jsonapi/service/processor/HybridFieldExpander.java

```diff
@@ -0,0 +1,103 @@
+package io.stargate.sgv2.jsonapi.service.processor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import io.stargate.sgv2.jsonapi.api.model.command.Command;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandContext;
+import io.stargate.sgv2.jsonapi.api.model.command.impl.InsertManyCommand;
+import io.stargate.sgv2.jsonapi.api.model.command.impl.InsertOneCommand;
+import io.stargate.sgv2.jsonapi.config.constants.DocumentConstants;
+import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+import io.stargate.sgv2.jsonapi.util.JsonUtil;
+import java.util.Iterator;
+
+/**
+ * Helper class that encapsulates the logic for expanding "$hybrid" fields in Collection Documents
+ * to insert.
+ *
+ * <p>Note that this expansion is done before checking whether either lexical or vectorize fields
+ * are actually supported -- this is verified at later point. Note, too, that input document(s) are
+ * modified in place, due to design of "$vectorize" support for Collections.
+ */
+public class HybridFieldExpander {
+  public static void expandHybridField(CommandContext context, Command command) {
+    // Only support expansion for Collections
+    if (context.isCollectionContext()) {
+      // and just for Insert commands, in particular
+      switch (command) {
+        case InsertOneCommand cmd -> expandHybridField(0, 1, cmd.document());
+        case InsertManyCommand cmd -> {
+          var docs = cmd.documents();
+          if (docs != null) {
+            for (int i = 0, len = docs.size(); i < len; ++i) {
+              expandHybridField(i, len, docs.get(i));
+            }
+          }
+        }
+        default -> {}
+      }
+    }
+  }
+
+  // protected for testing purposes
+  protected static void expandHybridField(int docIndex, int docCount, JsonNode docNode) {
+    final JsonNode hybridField;
+
+    if ((docNode instanceof ObjectNode doc)
+        && (hybridField = doc.remove(DocumentConstants.Fields.HYBRID_FIELD)) != null) {
+      switch (hybridField) {
+        case NullNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
+        case TextNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
+        case ObjectNode ob -> addFromObject(doc, ob, docIndex, docCount);
+        default ->
+            throw ErrorCodeV1.HYBRID_FIELD_UNSUPPORTED_VALUE_TYPE.toApiException(
```

Comment:
in the FindAndRerankSortClauseDeserializer I was throwing the JsonMappingException when there was schema problem like this 

also , have a look at the JsonFieldMatcher use at the top of the class, I was trying to make this sort of deserialisation easier

Reply:
While possible, something would then need to map that exception to error code? Seems simpler to produce actual error code we want right where we know it.
Will have a look at `FindAndRerankSortClauseDeserializer`

Reply:
Since this is not a Jackson `JsonDeserializer`, it's easy to throw our own API errors, so it seems better to throw specific API error code -- otherwise we need another mapper for doing that, adding to complexity? (since we don't expose `JsonMappingException`s?

I can change this if we want to go the other way, just not sure of advantages of going through transformation route.